src/rust/engine/fs/fs_util/src/main.rs

```diff
@@ -480,7 +486,40 @@ async fn execute(top_match: &clap::ArgMatches<'_>) -> Result<(), ExitError> {
           .unwrap()
           .parse::<usize>()
           .expect("size_bytes must be a non-negative number");
-        let digest = Digest(fingerprint, size_bytes);
+        let mut digest = Digest(fingerprint, size_bytes);
+
+        if let Some(child_dir) = args.value_of("child-dir") {
+          let mut accumulated_components = PathBuf::new();
+          for component in Path::new(child_dir) {
+            if component.is_empty() {
+              continue;
+            }
+            accumulated_components.push(component);
+            let (directory, _) = store.load_directory(digest).await?.ok_or_else(|| {
+              format!(
+                "Couldn't find digest {:?} for child dir {:?}",
+                digest, accumulated_components
+              )
+            })?;
+            let mut found_child = false;
+            for directory in directory.get_directories() {
+              if directory.get_name() == component {
+                let maybe_digest: Result<Digest, String> = directory.get_digest().try_into();
+                digest = maybe_digest.map_err(|err| {
+                  format!(
+                    "Digest of child dir {:?} couldn't be parsed: {:?}",
+                    accumulated_components, err
+                  )
+                })?;
+                found_child = true
+              }
+            }
+            if !found_child {
+              panic!("Didn't find child dir {:?}", accumulated_components);
+            }
+          }
```

Comment:
Consider adding this to `SnapshotOps` instead? https://github.com/pantsbuild/pants/blob/239f8879f09ebc7c4b859bfa0a59b6906f8e570c/src/rust/engine/fs/store/src/snapshot_ops.rs#L645-L652

Reply:
Re-implemented in terms of `SnapshotOps::strip_prefix` - PTAL!